scripts/istat/geos/preprocess.py

```diff
@@ -1,96 +1,103 @@
 
+import zipfile
+import io
+import os
+import requests
 import pandas as pd
 
-FILE_PATH_it = "./Elenco-codici-statistici-e-denominazioni-al-01_01_2020_it.csv"
-FILE_PATH_en = "./Elenco-codici-statistici-e-denominazioni-al-01_01_2020_en.csv"
+URL = "https://www.istat.it/storage/codici-unita-amministrative/"+\
+    "Elenco-codici-statistici-e-denominazioni-delle-unita-territoriali.zip"
+
+def download(url):
+    """download and extract the csv file to ./raw/xx"""
+    response = requests.get(url)
+    file_zip = zipfile.ZipFile(io.BytesIO(response.content))
+    file_zip.extractall("./raw/temp")
+    folder_name = os.listdir("./raw/temp")[0]
+    folder = os.path.join("./raw/temp", folder_name)
+    files = [file for file in os.listdir(folder) if file[-4:] == ".csv"]
+    if len(files) != 1:
+        raise Exception("0 or more than 1 files found")
+    file_name = files[0]
+    os.rename(os.path.join(folder, file_name), "./raw/"+file_name[:-4]+"_it.csv")
+    os.system("rm -rf ./raw/temp")
+    return "./raw/"+file_name[:-4]+"_it.csv"
+
+def translate(file_path):
+    """tranlate the column names to English based on Google translate"""
+    data = pd.read_csv(file_path, sep=';', encoding='cp1252')
+    data.columns = ["Region Code", "Supra-municipal territorial unit"+\
+        "code (valid for statistical purposes)", "Province Code (Historic) (1)",  \
+        "Municipality progress (2)", "Common alphanumeric format code", \
+        "Name (Italian and foreign)", "Name in Italian", "Name in other language",\
+         "Geographic breakdown code", "Geographical breakdown", "Region name", \
+         "Name of the supra-municipal territorial unit (valid for statistical purposes)", \
+        "Flag Municipality provincial capital / metropolitan city / free consortium", \
+        "Automotive abbreviation", "Common Code numeric format", \
+        "Numeric Common Code with 110 provinces (from 2010 to 2016)", \
+        "Numeric Common Code with 107 provinces (from 2006 to 2009)", \
+        "Numerical Common Code with 103 provinces (from 1995 to 2005)", \
+        "Cadastral code of the municipality", \
+        "Legal population 2011 (09/10/2011)", "NUTS1", "NUTS2(3)", "NUTS3"]
+    file_path_en = file_path.replace('_it.csv', '_en.csv')
+    data.to_csv(file_path_en, encoding='utf-8', index=False)
+    return file_path_en
+
+def preprocess(file_path):
+    """"preprocess the csv file for importing into Data Commons"""
+    data = pd.read_csv(file_path)
+    columns_rename = {"Province Code (Historic) (1)": "Province Code",\
+        "Name of the supra-municipal territorial unit (valid for statistical purposes)":\
+        "Province name", "Automotive abbreviation":"Province Abbreviation",\
+        "NUTS2(3)":"NUTS2", "Common Code numeric format":"Municipal Code", \
+        "Name in Italian":"Municipal Name"}
+    data = data.rename(columns=columns_rename)
+
+    # correct some of the mismatch of NUTS code and names
```

Comment:
Could you make some more detailed notes, either here, or in the README? Geos are sensitive and we want to make sure we document any hardcoding

Reply:
I added descriptions on how and why I made the changed in the comment of the script

Reply:
I think it's fine. Most of these come from a [reorg](https://screenshot.googleplex.com/oGT0MbR8N25).